packages/gatsby/src/schema/create-key.js

```diff
@@ -1,3 +1,3 @@
 const regex = new RegExp(`[^a-zA-Z0-9_]`, `g`)
 
-module.exports = key => key.replace(regex, `___`)
+module.exports = key => key.replace(`@`, ``).replace(regex, `___`)
```

Comment:
I don't think removing this is the best option, you can get namespace conflcits on fields without the `@` that aren't resolvable.

I did fix this in two ways tho in #1136